src/app/main.rs

```diff
@@ -298,6 +303,7 @@ where
                         .clone()
                         .await_crt()
                         .map(move |id| {
+                            drop(ready_latch);
```

Comment:
TIOLI: the fact that dropping the latch indicates readiness seems a little non-obvious without actually reading the `ready` module.

What about adding a function like 
```rust
impl Latch {
    pub fn set_ready(self) {
        drop(self);
    }
}
```
so that this could be `ready_latch.set_ready()` instead? 

Admittedly, this adds some more code that isn't _strictly_ necessary, but it communicates the semantic meaning of this drop a little better. Up to you!

Reply:
yeah that seems fine. the main goal was to _require_ acting on the latch to signal readiness. if you drop them all, it should just be ready. but this is clearer as a golden-path api.